packages/gcf-utils/src/logging-octokit-plugin.ts

```diff
@@ -0,0 +1,245 @@
+      owner: string;
+    };
+  };
+}
+
+/**
+ * Parsed details of GitHub action
+ */
+interface GitHubActionDetails {
+  type?: GitHubActionType;
+  value?: string;
+  destObjType?: GitHubObjectType;
+  destObjId?: string | number;
+  repoName?: string;
+  repoOwner?: string;
+}
+
+/**
+ * A logger with a metric endpoint
+ */
+interface MetricLogger {
+  metric: {(data: GitHubActionLog): void};
+}
+
+/**
+ * Maps GitHub API endpoints to a GitHubActionType
+ */
+const ActionEndpoints: {[url: string]: {[method: string]: GitHubActionType}} = {
+  '/repos/:owner/:repo/issues/:issue_number/labels': {
+    POST: GitHubActionType.ISSUE_ADD_LABELS,
+  },
+  '/repos/:owner/:repo/labels': {
+    POST: GitHubActionType.ISSUE_CREATE_LABEL,
+  },
+  '/repos/:owner/:repo/issues/:issue_number/comments': {
+    POST: GitHubActionType.ISSUE_CREATE_COMMENT,
+  },
+  '/repos/:owner/:repo/issues/:issue_number/labels/:name': {
+    DELETE: GitHubActionType.ISSUE_REMOVE_LABEL,
+  },
+  '/repos/:owner/:repo/labels/:name': {
+    DELETE: GitHubActionType.ISSUE_DELETE_LABEL,
+  },
+  '/repos/:owner/:repo/labels/:current_name': {
+    PATCH: GitHubActionType.ISSUE_UPDATE_LABEL,
+  },
+  '/repos/:owner/:repo/issues/:issue_number': {
+    PATCH: GitHubActionType.ISSUE_UPDATE,
+  },
+  '/repos/:owner/:repo/issues': {
+    POST: GitHubActionType.ISSUE_CREATE,
+  },
+  '/repos/:owner/:repo/pulls/:pull_number/reviews/:review_id/dismissals': {
+    PUT: GitHubActionType.PR_DISMISS_REVIIEW,
+  },
+  '/repos/:owner/:repo/pulls/:pull_number/merge': {
+    PUT: GitHubActionType.PR_MERGE,
+  },
+  '/repos/:owner/:repo/pulls/:pull_number/update-branch': {
+    PUT: GitHubActionType.PR_UPDATE_BRANCH,
+  },
+};
+
+function parseActionValue(
+  actionType: GitHubActionType,
+  options: {
+    [key: string]: string | number;
+  }
+): string {
+  // optional properties for issues.update
+  const allIssueUpdateProps = [
+    'body',
+    'state',
+    'assignees',
+    'labels',
+    'milestone',
+    'title',
+  ];
+  const hasIssueUpdateProps = allIssueUpdateProps.filter(
+    (prop: string) => options[prop]
+  );
+
+  switch (actionType) {
+    case GitHubActionType.ISSUE_ADD_LABELS:
+      return String(options.labels);
+    case GitHubActionType.ISSUE_CREATE_COMMENT:
+      return String(options.body);
+    case GitHubActionType.ISSUE_REMOVE_LABEL:
+    case GitHubActionType.ISSUE_CREATE_LABEL:
+    case GitHubActionType.ISSUE_DELETE_LABEL:
+      return String(options.name);
+    case GitHubActionType.ISSUE_UPDATE_LABEL:
+      return `${options.current_name} to ${options.name}`;
+    case GitHubActionType.ISSUE_UPDATE:
+      return 'updated: ' + hasIssueUpdateProps.join(',');
+    case GitHubActionType.ISSUE_CREATE:
+      return String(options.title);
+    case GitHubActionType.PR_DISMISS_REVIIEW:
+      return `dismiss ${options.review_id}: ${options.message}`;
+    case GitHubActionType.PR_MERGE:
+    case GitHubActionType.PR_UPDATE_BRANCH:
+    default:
+      return 'NONE';
```

Comment:
You mean the `NONE` value? The switch here is returning the `value` property based on `GitHubActionType` so `NONE` here refers to the `value` rather than the type. I already have an `UNKNOWN` item in `GitHubActionType. Does that answer your question or am I misunderstanding?

Reply:
It just seems a bit odd to have every other option return a `GitHubActionType` enum value (which is a string) yet `NONE` is just a string. It's possible to make this method return `GitHubActionType` if we want to (and if it makes sense).